client/src/pages/editor.js

```diff
@@ -23,14 +23,40 @@ function getSchemaForSchemaEditor(schema) {
   return schema;
 }
 
-@inject(ItemStore, MessageService, DialogService, I18N)
+function getTranslatedSchema(schema, toolName, i18n) {
+  schema = JSON.parse(JSON.stringify(schema));
+  if (schema.title) {
+    schema.title = i18n.tr(`${toolName}:${schema.title}`);
+  }
+  if (schema.hasOwnProperty('items')) {
+    if (schema.items.hasOwnProperty('oneOf')) {
+      schema.items.oneOf = schema.items.oneOf.map(oneOfSchema => getTranslatedSchema(oneOfSchema));
+    } else {
+      schema.items = getTranslatedSchema(schema.items, toolName, i18n);
+    }
+  }
+  if (schema.hasOwnProperty('properties')) {
+    Object.keys(schema.properties).forEach(propertyName => {
+      schema.properties[propertyName] = getTranslatedSchema(schema.properties[propertyName], toolName, i18n);
+    });
+  }
+  if (schema.hasOwnProperty('Q:options') && schema['Q:options'].hasOwnProperty('enum_titles')) {
+    schema['Q:options'].enum_titles = schema['Q:options'].enum_titles.map(title => {
+      return i18n.tr(`${toolName}:${title}`);
+    });
+  }
+  return schema;
+}
+
+@inject(ItemStore, MessageService, DialogService, I18N, EventAggregator)
 export class Editor {
 
-  constructor(itemStore, messageService, dialogService, i18n) {
+  constructor(itemStore, messageService, dialogService, i18n, ea) {
     this.itemStore = itemStore;
     this.messageService = messageService;
     this.dialogService = dialogService;
     this.i18n = i18n;
+    this.ea = ea;
```

Comment:
I would call it `eventAggregator`, so it's clear what's meant in code sections which are further below too.